homeassistant/components/xiaomi_miio/vacuum.py

```diff
@@ -209,22 +218,22 @@ async def async_setup_entry(hass, config_entry, async_add_entities):
 class MiroboVacuum(XiaomiMiioEntity, StateVacuumEntity):
     """Representation of a Xiaomi Vacuum cleaner robot."""
 
-    def __init__(self, name, device, entry, unique_id):
+    def __init__(
+        self, name, device, entry, unique_id, coordinator: DataUpdateCoordinator
+    ):
         """Initialize the Xiaomi vacuum cleaner robot handler."""
         super().__init__(name, device, entry, unique_id)
 
         self.vacuum_state = None
```

Comment:
Consider removing this attribute and just access the coordinator.data instead in the properties.